### PR TITLE
Default GraphBasedChecking, ParallelOptimization and ParallelIlxGen optimizations

### PR DESCRIPTION
**Pull request by edgarfgp**
## Description

Enable by default:

- GraphBasedChecking
- ParallelOptimization 
- ParallelIlxGen

## Checklist

- [ ] Test cases added
- [x] Release notes entry updated

### CODE REVIEW

File: tests/FSharp.Compiler.ComponentTests/TypeChecks/Graph/CompilationTests.fs [hunk 1]
```diff
@@ -13,7 +13,7 @@ type Method =
 let methodOptions (method: Method) =
     match method with
     | Method.Sequential -> []
-    | Method.Graph -> [ "--test:GraphBasedChecking"; "--test:DumpCheckingGraph" ]
+    | Method.Graph -> [ "GraphBasedChecking"; "DumpCheckingGraph" ]
```
Comment (nojaf) [inline tests/FSharp.Compiler.ComponentTests/TypeChecks/Graph/CompilationTests.fs:16]: Maybe keep `--test:DumpCheckingGraph` as test flag? It still very much is a contributor debug thing.

Comment (vzarytovskii) [reply tests/FSharp.Compiler.ComponentTests/TypeChecks/Graph/CompilationTests.fs:16]: Yeah, can't remove them due to backwards compatibility

Comment (edgarfgp) [reply tests/FSharp.Compiler.ComponentTests/TypeChecks/Graph/CompilationTests.fs]: Done. Thanks

**Comment by edgarfgp** [comment]
@vzarytovskii @KevinRansom Would like to propose to enable this optimization by default. I'm using Kevin's Defaultrealsig[PR](https://github.com/dotnet/fsharp/pull/17385/files) as an example. Let me know your thoughts and any pointer is appreciated.

**Comment by vzarytovskii** [comment]
Could you please mention in FCS changelog that we changed the defaults?

**Comment by edgarfgp** [comment]
> Could you please mention in FCS changelog that we changed the defaults?

Done. Not sure why the bot decided to put the `No release notes required`

File: tests/FSharp.Compiler.ComponentTests/EmittedIL/Inlining/Match01_RealInternalSignatureOff.fs.il.netcore.release.bsl [hunk 2]
```diff
@@ -1107,6 +1099,42 @@
     } 
   } 
 
+  .method public static int32  select1(class Match01/Test1 x) cil managed
```
Comment (vzarytovskii) [inline tests/FSharp.Compiler.ComponentTests/EmittedIL/Inlining/Match01_RealInternalSignatureOff.fs.il.netcore.release.bsl:1102]: That doesn't look good, that it changed ordering. A potential determinism issue here

File: tests/FSharp.Compiler.ComponentTests/EmittedIL/TestFunctions/TestFunction17.fs.il.netcore.debug.bsl [hunk 3]
```diff
@@ -124,18 +116,12 @@
                int32 V_3,
                class [runtime]System.Collections.IComparer V_4,
                int32 V_5,
```
Comment (vzarytovskii) [inline tests/FSharp.Compiler.ComponentTests/EmittedIL/TestFunctions/TestFunction17.fs.il.netcore.debug.bsl:118]: Also doesn't look good, unsure why baselines changed, ilx output should retain the same order

**Pull request closed without merging**